test/s3select_test.cpp

```diff
@@ -1795,6 +1820,18 @@ TEST(TestS3selectFunctions, tuefalse_like_expressions)
   std::string s3select_result_8 = run_s3select(input_query_8,input);
 
   ASSERT_EQ(s3select_result_7, s3select_result_8);
+
+  const std::string input_query_9 = "select count(*) from stdin where (_1 like \"%r[r-s]\") = false;";
+
+  std::string s3select_result_9 = run_s3select(input_query_9,input);
+
+  ASSERT_NE(s3select_result_9,failure_sign);
+
+  const std::string input_query_10 = "select count(*) from stdin where (substring(_1,char_length(_1),1) not between \"r\" and \"s\") = true or (substring(_1,char_length(_1)-1,1) = \"r\") = false;";
+
+  std::string s3select_result_10 = run_s3select(input_query_10,input);
+
+  ASSERT_EQ(s3select_result_9, s3select_result_10);
```

Comment:
please check this test, the results are empty.

Reply:
Strange, I am getting 10k for both. I guess the current data set used for these queries doesn't contain those patterns

Reply:
the query should not return also the full data set (the query is not distinctive enough).

it's quite easy to verify results.

Reply:
ok, it returns results 
what was the change?

Reply:
ok, got that.